assets/js/frontend/checkout.js

```diff
@@ -426,14 +428,21 @@ jQuery( function( $ ) {
 						} catch ( e ) {
 
 							// Attempt to fix the malformed JSON
-							var valid_json = raw_response.match( /{"result.*"}/ );
+							var maybe_valid_json = raw_response.match( /{"result.*}/ );
 
-							if ( null === valid_json ) {
+							if ( null === maybe_valid_json ) {
 								console.log( 'Unable to fix malformed JSON' );
 							} else {
-								console.log( 'Fixed malformed JSON. Original:' );
-								console.log( raw_response );
-								raw_response = valid_json[0];
+								// Check for valid JSON
+								data = $.parseJSON( maybe_valid_json[0] );
```

Comment:
Yeah, I think I'd just make a function that validates JSON, catches exception and returns true/false. Then you could use it both places.